src/NJsonSchema.CodeGeneration.CSharp/Templates/ClassTemplate.tt

```diff
@@ -5,6 +5,12 @@
 <#if(Model.HasDiscriminator){#>
 [Newtonsoft.Json.JsonConverter(typeof(JsonInheritanceConverter), "<#=Model.Discriminator#>")]
 <#}#>
+<#if(Model.HasCustomName){#>
+[NJsonSchema.Annotations.JsonSchema("<#=Model.CustomName#>")]
```

Comment:
We shouldn't add NJsonSchema dependent attributes here... The code should work out-of-the-box.